### PR TITLE
Implement proofread filter functionality to Browse Chants page

### LINKED ISSUE #1492: Proofreaders need some easier way to see their "task list" (chants that need proofreading)
In discussions with Debra:

- could this be like a task queue?
- could this be just another sortable column on the source inventory page that is visible only to proofreaders of the source?

### PR DESCRIPTION
**Pull request by lucasmarchd01**
This PR introduces functionality for filtering by proofreading status on the "Browse Chants" page. It includes:

- Filtering functionality for `manuscript_full_text_proofread`, `manuscript_full_text_std_proofread`, and `volpiano_proofread fields`.
- JavaScript updates to update URL parameters dynamically as the user selects different proofreading options.
- Forms with ChoiceField and RadioSelect widgets for filtering chants based on proofreading status.
- User permission checks to determine if a user can proofread a specific source.
- Templates are updated to reflect the new proofreading filter options.
- Add a test suite for these changes

Updated screenshot:
image

Towards #1492

### CODE REVIEW

**Comment by lucasmarchd01** [comment]
Just adding another thought here...
I implemented this on the Source Browse Chants page. However, it could be on the source inventory page instead (or both). My thought was that since the browse chants page also has the "Edit" links it would be easier for a proofreader to filter and edit these chants directly. But if this looks to cluttered or belongs on another page we can definitely switch that before merging.

File: django/cantusdb_project/main_app/permissions.py [hunk 1]
```diff
@@ -61,6 +61,28 @@ def user_can_proofread_chant(user: User, chant: Chant) -> bool:
     return user_is_project_manager or (user_is_editor and user_is_assigned_to_source)
 
 
+def user_can_proofread_source(user: User, source: Source) -> bool:
```
Comment (dchiller) [inline django/cantusdb_project/main_app/permissions.py:68]: If you've written this function, the function right above could make use of it!

File: django/cantusdb_project/main_app/templates/browse_chants.html [hunk 2]
```diff
@@ -58,6 +58,21 @@ <h3>Browse Chants</h3>
                 </select>
             </div>
         </div>
+        {% if user_can_proofread_source %}
```
Comment (dchiller) [inline django/cantusdb_project/main_app/templates/browse_chants.html:61]: I find the radio buttons really close together with the text -- any chance the options could be more spatially differentiated?

File: django/cantusdb_project/main_app/views/source.py [hunk 3]
```diff
@@ -84,8 +93,17 @@ def get_queryset(self):
...
+            "manuscript_full_text_proofread"
+        )
+        manuscript_full_text_std_proofread = self.request.GET.get(
+            "manuscript_full_text_std_proofread"
+        )
+        volpiano_proofread = self.request.GET.get("volpiano_proofread")
+
         # get all chants in the specified source
-        chants = source.chant_set.select_related("feast", "service", "genre")
+        chants: QuerySet = source.chant_set.select_related("feast", "service", "genre")
```
Comment (dchiller) [inline django/cantusdb_project/main_app/views/source.py]:
```suggestion
        chants: QuerySet[Chant] = source.chant_set.select_related("feast", "service", "genre")
```

**Review by dchiller — APPROVED** [review]
What's here look good, but I think it would be to not close #1492 yet, because what's here is useful for seeing what chants in a source need proofreading, but you have to know which sources have these chants to make it useful. I'm not sure if that's always the case? Maybe @annamorphism can let us know based on experience. If it's helpful to see this at the source level too, we may want to add a similar function to the Source List page.

File: django/cantusdb_project/main_app/templates/browse_chants.html (hunk 2 above)
Comment (annamorphism) [reply django/cantusdb_project/main_app/templates/browse_chants.html:61]: > I find the radio buttons really close together with the text -- any chance the options could be more spatially differentiated?

I agree--I think it's that the buttons before and after "Yes" are both about equally close...

**Comment by annamorphism** [comment]
> What's here look good, but I think it would be to not close #1492 yet, because what's here is useful for seeing what chants in a source need proofreading, but you have to know which sources have these chants to make it useful. I'm not sure if that's always the case? Maybe @annamorphism can let us know based on experience. If it's helpful to see this at the source level too, we may want to add a similar function to the Source List page.

@dchiller yes, it's not always known at the source level if a source is "done" or not--for example, I know I've proofread *some* fragments from Trier and not others. This way I can go in and check, but a little flag that "this source has some chants that need proofreading" would help find them.

**Pull request merged**